### PR TITLE
[macOS] Remove deprecated altool notarization support, disable rcodesign for C# version.

### PR DESCRIPTION
**Pull request by bruvzg**
- Removes support for deprecated `altool` notarization.
- Disables `rcodesign` support for C# version and adds an error message if GDExtension libraries are found during export, since `rcodesign` is not capable of signing bundles with embedded libraries.

### CODE REVIEW

**Review by akien-mga — APPROVED** [review]

File: platform/macos/export/export_plugin.cpp [hunk 1]
```diff
@@ -395,9 +392,17 @@ void EditorExportPlatformMacOS::get_export_options(List<ExportOption> *r_options
 	r_options->push_back(ExportOption(PropertyInfo(Variant::STRING, "xcode/xcode_build"), "14C18"));
 
 #ifdef MACOS_ENABLED
-	r_options->push_back(ExportOption(PropertyInfo(Variant::INT, "codesign/codesign", PROPERTY_HINT_ENUM, "Disabled,Built-in (ad-hoc only),rcodesign,Xcode codesign"), 3, true));
+	if (Engine::get_singleton()->has_singleton("GodotSharp")) {
+		r_options->push_back(ExportOption(PropertyInfo(Variant::INT, "codesign/codesign", PROPERTY_HINT_ENUM, "Disabled,Built-in (ad-hoc only),Xcode codesign:3"), 3, true));
```
Comment (raulsntos) [inline platform/macos/export/export_plugin.cpp:396]: What about existing export presets that are already set to `2`?

**Review by raulsntos — COMMENTED** [review]
What does it mean to disable rcodesign for C#? Is that going to cause issues for exported projects? Should there be a warning in this case too?

Also, we should support exporting non-C# projects from the C# version of the editor. What does this change mean for this case?

File: platform/macos/export/export_plugin.cpp (hunk 1 above)
Comment (akien-mga) [reply platform/macos/export/export_plugin.cpp:396]: Maybe it's simpler to keep the same enum for all builds, but just show the error added below if a user selects rcodesign for a C# project. It's not the best UX but it should simplify things and not be that bad either (at least not worse than a lot of the export settings UX).

Comment (bruvzg) [reply platform/macos/export/export_plugin.cpp:396]: Changed it to a warning.

**Review by akien-mga — APPROVED** [review]

**Comment by YuriSizov** [comment]
Thanks!

**Pull request merged**